app/controllers/admin/settings/base_controller.rb

```diff
@@ -1,37 +1,21 @@
 module Admin
   module Settings
     class BaseController < Admin::ApplicationController
-      MISMATCH_ERROR = "The confirmation key does not match".freeze
-
-      before_action :extra_authorization_and_confirmation, only: [:create]
+      before_action :authorize_super_admin
 
       def create
         result = upsert_config(settings_params)
 
         if result.success?
           Audit::Logger.log(:internal, current_user, params.dup)
-          redirect_to admin_config_path, notice: "Successfully updated settings."
+          render json: { message: "Successfully updated settings." }, status: :ok
         else
-          redirect_to admin_config_path, alert: "😭 #{result.errors.to_sentence}"
+          render json: { error: result.errors.to_sentence }, status: :unauthorized
```

Comment:
I'm don't think `:unauthorized` is the right status here. You call `authorize_super_admin` before we get here, so at this point, the update failed for other reasons. Usually we use `:unprocessable_entity` I think.

Reply:
Yes I hadn’t quite gotten to changing it 😅